frontend/src/hooks/useSelectAllContent.ts

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import type { RefObject } from "react";
+import { useEventListener } from "./useEventListener";
+
+/**
+ * Hook that enables Ctrl/Cmd-A to select all text content within a specific element
+ * @param ref - React ref to the element whose content should be selectable
+ * @param enabled - Whether the hook should be active (defaults to true)
+ */
+export function useSelectAllContent<T extends HTMLElement>(
+  ref: RefObject<T | null>,
+  enabled = true,
+) {
+  const handleKeyDown = (event: KeyboardEvent) => {
```

Comment:
Would we want a useCallback for this? Not sure if `enabled` is dynamic.

Reply:
we use react-compiler so don't need `useCallback` anymore. also the `useEventHandler` does handle keeping a stable ref